src/cloudant/result.py

```diff
@@ -358,6 +360,8 @@ def __iter__(self):
             if result:
                 for row in result:
                     yield row
+                if len(result) < self._page_size:
```

Comment:
We still do an unnecessary request if the total number of view results == `self._page_size` I think. I'm wondering if we can check total_rows against self._page_size too?

Reply:
I think it's safer for now to keep the double request when `results == page_size` than trying to use `total_rows`.